File: src/Adapter/Scala/HeadsetAdapter.scala
package Adapter.Scala

/**
  * Created by prefert on 2017/11/20.
  */
class HeadsetAdapter extends PhoneJackConnector with Lightning {
  override def audioTraditionally: Unit = println("通过 Lightning 播放声音")

  override def audioWithLightning: Unit = println("通过 Lightning 播放声音")
}

File: src/Adapter/Java/HeadsetAdapter.java
package Adapter.Java;

/**
 * Created by prefert on 2017/11/19.
 * 耳机适配器 传统 => lightning
 */
public class HeadsetAdapter extends LightningConnector implements PhoneJackInterface {
    @Override
    public void audioTraditionally() {
        // 传统接口兼容 lightning
        super.audioWithLightning();
    }
}
